app/services/stats/calculations.rb

```diff
@@ -160,32 +160,58 @@ def self.populate_team_details_callback(stats)
         team = teams.detect { |t| t.id == team_id }
         case team.class.to_s
         when 'BusinessUnit'
-          result_set[:business_unit] = team.name
-          result_set[:directorate] = team.parent.name
-          result_set[:business_group] = team.parent.parent.name
+          business_unit_result_setter(result_set, team)
         when 'Directorate'
-          result_set[:business_unit] = ''
-          result_set[:directorate] = team.name
-          result_set[:business_group] = team.parent.name
+          directorate_result_setter(result_set, team)
         when 'BusinessGroup'
-          result_set[:business_unit] = ''
-          result_set[:directorate] = ''
-          result_set[:business_group] = team.name
+          business_group_result_setter(result_set, team)
         else
           raise "Invalid team type #{team.class}"
         end
-        result_set[:deactivated] = team.deactivated
-        result_set[:responsible] = team.team_leader_name
-        result_set[:moved] = team.moved
+        default_results_setter(result_set, team)
       end
+      stats_total_setter(stats)
+    end
+
+    def self.business_unit_result_setter(result_set, team)
+      result_set[:business_unit] = team.name
+      result_set[:business_unit_id] = team.id
+      result_set[:previous_business_unit_id] = team.moved_to_unit_id
```

Comment:
I think `moved_to_unit_id` is the new unit the current unit moved to, not the previous one,  it should be next one?